zowe/zowe-cli
# Enhancement: Add progress tracking to DirToUSSDirHandler for file upl…

**What It Does**  
This update enhances the `DirToUSSDirHandler` functionality:  
- Introduced progress tracking using `commandParameters.response.progress` to provide feedback during file uploads.  
- Reintroduced the `encoding` parameter in the upload options to ensure compatibility with various encoding configurations.  

fixed #2344 

**How to Test**  
1. Configure the handler with a valid session and set `encoding` in the upload options.  
2. Run the upload command with both `recursive` and non-recursive options.  
3. Verify that a progress bar is displayed during the upload process.  
4. Check the console log for properly formatted responses.  

**Review Checklist**  
I certify that I have:  
- [x] followed the [[contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)  

**Additional Comments**  
This update ensures better user feedback through progress tracking and retains support for encoding, making the handler more robust.
